### DISCUSSION TITLE
Configure Git LFS for enterprise...?

### DISCUSSION THREAD

**Discussion by carlbeechvuw** [community/community#126491] [Enterprise]
### Body

Hi
We're users of the Enterprise Edition (3.12.4) under the campus program as an VM appliance - we've had a number of users start to ask about committing larger files to github - and they would like to use git LFS. Some have already used it, and their data is stored in /data/user/storage - but having the data stored within the appliance means the use of 'expensive' storage - i.e. within our vm environment.

I've spotted 'Configuring Git Large File Storage to use a third party server' (https://docs.github.com/en/enterprise-server@3.10/admin/managing-accounts-and-repositories/managing-repositories-in-your-enterprise/configuring-git-large-file-storage-for-your-enterprise)

However, this appears to be on a per-repo basis - however, I thought there might be a way to configure this centrally within the enterprise edition?   - To all intents and purposes a simple NFS mount over /data/user/storage - however, NFS mounts are not supported in github enterprise...

Am I missing something, or does someone have a set of docs / suggestion as to how this should be handled?

We do have S3 compatible storage available...

Thanks
CArl.

**Comment by joshjohanning** [comment]
Unfortunately, the way that Git LFS works as a git extension requires everything to be set at the repository level. 

By default, LFS tries to post to the same URL as the main git server and appending `/api/lfs` to the endpoint. 

You can configure the endpoint by creating a `.lfsconfig` file in the root of the repo. 

You could make this file exist as part of an onboarding repo template (ie everything use this repo template when creating a new repo) or use a tool like [multi-gitter](https://josh-ops.com/posts/github-script-to-add-dependabot-file/) (examples in link) to push a file to a set of repos instead. 

Here’s an example of configuring LFS with Artifactory: https://community.atlassian.com/t5/Bitbucket-articles/Managing-LFS-Files-in-JFrog-Artifactory/ba-p/1603122

Note that whenever you have an external LFS endpoint you’ll need to manage credentials for it separately. Depending on how the user’s git environment is set up, it should prompt for credentials when running LFS commands against that endpoint for the first time.
